src/ripple/protocol/impl/STParsedJSON.cpp

```diff
@@ -872,7 +861,17 @@ STParsedJSONArray::STParsedJSONArray (
     Json::Value const& json)
 {
     using namespace STParsedJSONDetail;
-    parseArray (name, json, sfGeneric, 0, array, error);
+    auto arr = parseArray (name, json, sfGeneric, 0, error);
+    if (!arr)
+        array = boost::none;
+    else
+    {
+        auto p = dynamic_cast <STArray*> (&arr->get());
+        if (p == nullptr)
```

Comment:
How would `parseArray` return anything other than `boost::none` or an `STArray`? The only thing I can imagine is a bug, in which case, this check would be better as an `assert`.